client/abstract-client.js

```diff
@@ -205,11 +217,9 @@ class AbstractClient {
             throw Error("Unable to get results, need handler id");
         }
         let searchResponse = {};
-        let retries = 0;
-        let timeout = options.timeout ? options.timeout : this.defaultTimeoutInSeconds;
-        let numberOfResults = options.numberOfResults
-            ? options.numberOfResults
-            : this.defaultNumberOfResults;
+        let timeoutInSeconds = options.timeoutInSeconds ? options.timeoutInSeconds : this.defaultTimeoutInSeconds;
```

Comment:
I think that here should be this.timeoutInSeconds

Reply:
Instead of defaultTimeoutInSeconds?

Reply:
Yes, because we in constructor assign defaultTimeoutInSeconds to the this.timeoutInSeconds if there is no timeoutInSeconds provided in constructor, and then use it everywhere in code. For me, it makes sense, if developer does not provide any timeout the default will be setup in constructor and used everywhere, if he provides timeout in constructor that value will be used everywhere, and for every specific call if he provides the value that value will be used

Reply:
make sense, just to confirm